spec/units/frame_spec.rb

```diff
@@ -0,0 +1,221 @@
+require 'frame'
+
+describe Frame do
+
+  subject(:frame) { Frame.new(1) }
```

Comment:
This is not best practice. You do not want tests to rely on each other, or in the order they are written in. Otherwise, if you remove one tests, the others would fail. Instead, each test should be independent, and work whatever the order (some testing frameworks actually randomize test order).